src/com/facebook/buck/swift/SwiftCompile.java

```diff
@@ -402,7 +404,7 @@ public SourcePath getSourcePathToOutput() {
             .toToolFlags(
                 resolver,
                 PathShortener.byRelativizingToWorkingDir(getProjectFilesystem().getRootPath()),
-                CxxDescriptionEnhancer.frameworkPathToSearchPath(cxxPlatform, resolver),
+                frameworkPathToSearchPath,
```

Comment:
According to the comment above:
>  Arg list can't simply be passed in since the current implementation of toToolFlags drops the
> dependency information.

I don't like the idea of making `frameworkArgs.frameworkPathToSearchPath` public, so I chose to pass it in directly instead and marked it as `@AddToRuleKey`.

However, this diminishes the benefit of having `frameworkArgs`. And it's the other reason why I think it might make sense to not use that new `frameworkArgs`, but to use `Function<FrameworkPath, Path> frameworkPathToSearchPath` and `ImmutableSortedSet<FrameworkPath> frameworkPaths` directly instead.

Reply:
@xianwen:
> I don't like the idea of making frameworkArgs.frameworkPathToSearchPath public, so I chose to pass it in directly instead and marked it as @AddToRuleKey

Agreed, it should not be exposed.

> Arg list can't simply be passed in since the current implementation of toToolFlags drops the dependency information.

I have a suspicion that this comment might be out of date (it was added back in 2017) and `CxxToolFlags` is annotated with `AddsToRuleKey`.

If we wanted to do this properly, we would pass `argsFromDeps` to the constructor, together with a single `Arg` for the frameworks. But it's safer to keep the current behaviour as changing it would require further validation.